plugins/modules/system/aix_devices.py

```diff
@@ -294,7 +294,7 @@ def main():
     module = AnsibleModule(
         argument_spec=dict(
             attributes=dict(type='dict'),
-            device=dict(type='str'),
+            device=dict(type='str', required=True),
```

Comment:
This is a breaking change. The code does handle a missing `device` in some cases.

Please undo this change, and fix the documentation instead (`required` needs to be removed there). It would be even better to properly document when `device` is needed and when not, but that's probably a lot of extra work.

Reply:
@felixfontein There's a couple of things to unpack here:
- there are 3 possible values for `state` as defined in `choices`, but the code and docs mention *aliases* to some of these states - this code needs a good revamp. I like the idea of doing that kind of work, but I prefer to focus on the validation-modules issue for the time being, and I have no AIX box available to test this on.
- despite the many actual or alleged options for state, the code is separated in two blocks: (available, present) and (removed, absent, defined).
  - In the "available" case, the first thing it does is to pass the device to the `lsdev -l <device>` command and, if the device does not exist, it moans and quits. Per docs (one quick search, YMMV) https://www.ibm.com/support/knowledgecenter/ssw_aix_71/l_commands/lsdev.html the argument to `-l` is mandatory, so we cannot have an empty string, nor I believe python's `None` will cut it.
  - In the  "removed" case, the first thing it does is check whether "device" is empty, if so, moans and quits.

Thus, I believe this is **not** a breaking change, we're just consolidating the existing logic into the existing feature of `AnsibleModule`. This module could use a little bit of a revamp, I wonder if @kairoaraujo is still working on it.

Reply:
I beg to differ, this is not a breaking change. The code actually requires a device parameter to exist (it cannot be empty or python's `None`).

But, for consistency's sake, as in other files, I will roll this back and another PR should be raised for that later on. Right now I want to keep the focus in removing as many ignore lines as possible.

Reply:
yeah, I was one reload short in the browser, that's why I wrote that again. :P anyways, I have reverted that change already